src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -600,12 +600,30 @@ func (c *controller) UploadLogs(ctx context.Context, wg *sync.WaitGroup, status
 				}
 				podName = pods[0].Name
 			}
-			err := common.UploadPodLogs(c.kc, c.ic, c.ClusterID, podName, c.Namespace, controllerLogsSecondsAgo, c.log)
+			var err error
+			c.log.Info("Collecting Logs ....")
+			if errorDetected != status.HasError() {
```

Comment:
sorry not sure i understand the logic. 
I thought we will have something more clear. 
if  status.HasError() {
  upload all logs and exit
}

Reply:
i think even better to just set ctx to Done and exit as expected without adding all this logic.
if status.HasError() {
  set done and let regular flow and exit
}

Reply:
It's possible and certainly ends up in more concise code, but it means that we will not have other logs from the routines that still waits. Is this OK?

Reply:
Also, In case of an error the context will be deleted twice (once from within UploadLogs and the second one from main). The second attempt will panic

Reply:
Fixed